kotlintest-assertions/src/main/kotlin/io/kotlintest/properties/Gen.kt

```diff
@@ -46,7 +47,8 @@ class BigIntegerGen(maxNumBits: Int) : Gen<BigInteger> {
  * The [Int] generator example should return a random int
  * from across the entire integer range.
  */
-interface Gen<T> {
+@higherkind
```

Comment:
We can HK interfaces now?

Reply:
In arrow we construct Monoid manually: https://github.com/arrow-kt/arrow/blob/master/modules/core/arrow-typeclasses/src/main/kotlin/arrow/typeclasses/Monoid.kt#L3-L5

Reply:
@pakoito AFAIK we were able to before but not sure about that particular case for Monoid

Reply:
You manually added the encoding there probably because you declare extensions in the same module as `@higherkind`, that was the actual limitation, we have not figured out annotation order with kapt.